packages/oruga/src/components/select/examples/base.vue

```diff
@@ -13,56 +20,18 @@ const selectedOptions = ref([]);
             </o-select>
         </o-field>
 
-        <o-field label="Grouped">
-            <o-select placeholder="Select a character" icon="user">
-                <optgroup label="Black Sails">
-                    <option value="flint">Flint</option>
-                    <option value="silver">Silver</option>
-                    <option value="vane">Vane</option>
-                    <option value="billy">Billy</option>
-                    <option value="jack">Jack</option>
-                </optgroup>
-
-                <optgroup label="Breaking Bad">
-                    <option value="heisenberg">Heisenberg</option>
-                    <option value="jesse">Jesse</option>
-                    <option value="saul">Saul</option>
-                    <option value="mike">Mike</option>
-                </optgroup>
-
-                <optgroup label="Game of Thrones">
-                    <option value="tyrion-lannister">Tyrion Lannister</option>
-                    <option value="jamie-lannister">Jamie Lannister</option>
-                    <option value="daenerys-targaryen">
-                        Daenerys Targaryen
-                    </option>
-                    <option value="jon-snow">Jon Snow</option>
-                </optgroup>
-            </o-select>
-        </o-field>
-
         <o-field
             label="Error"
             variant="danger"
             message="Something went wrong with this field">
-            <o-select placeholder="Select a character">
-                <option value="flint">Flint</option>
-                <option value="silver">Silver</option>
-            </o-select>
-        </o-field>
-
-        <o-field label="Rounded">
-            <o-select placeholder="Select a character" rounded>
-                <option value="flint">Flint</option>
-                <option value="silver">Silver</option>
-            </o-select>
+            <o-select placeholder="Select a character" :options="options" />
```

Comment:
Moving forward, is the `options` prop the recommended way to specify options for an `OSelect`? I'm not particularly opposed to that, as it provides a cleaner way to map between the "logical" property values and the strings used as the actual option values in the DOM. It does feel a little odd to have two different ways to specify the options, each of which has different functionality available (mapping between strings and arbitrary values for the `options` prop, grouping and more direct DOM control for `<option>` elements); perhaps the long-term goal should be to unify those somehow.

Reply:
Personally, I like to leave the rendering part to the component and just provide the data. So I added the `options` prop. However, there is no preferred way, either using the `options` prop or passing HMLT option elments inside the default slot.
You can pass any value when using the `options` prop, just like when using the HTML option.

It's kinda like other components; like the Table component for example, where you can provide the TableColumn components with the default slot or pass them as objects via the Table `data` prop.

Reply:
Vue extends the value type of select options to also support `objects` (https://vuejs.org/guide/essentials/forms.html#select-options). So, in my opinion, there's not that much difference between defining the options as a prop or as html elements.

Reply:
I kind of think Vue's choice there is a bit of a mistake, but I guess it does sort of work that way. The actual `<option>` values in the DOM are pretty weird (all set to the string `"[object Object]"`, at least in Firefox), which I don't really care for, but Vue does seem to patch things up so it still works.

Now that I look more closely, `OptionItem` doesn't have a separate string value for the `<option>` element, so my comment about that mapping is probably irrelevant.